lsp-types/src/Language/LSP/Protocol/Types/CustomInstances.hs

```diff
@@ -0,0 +1,27 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Language.LSP.Protocol.Types.CustomInstances where
+
+import           Language.LSP.Protocol.Internal.Types
+import           Data.Default
+import           Data.Hashable
+import           Data.Semigroup                ()
+
+instance Semigroup WorkspaceEdit where
+  (WorkspaceEdit a b c) <> (WorkspaceEdit a' b' c') = WorkspaceEdit (a <> a') (b <> b') (c <> c')
+instance Monoid WorkspaceEdit where
+  mempty = WorkspaceEdit Nothing Nothing Nothing
+
+instance Hashable Location
+instance Hashable Range
+instance Hashable Position
+
+instance Default ClientCapabilities where
+    def = ClientCapabilities (Just def) (Just def) (Just def) (Just def) (Just def) Nothing
```

Comment:
kinda suspicious that we don't use Nothing for the default values here?

Reply:
Looking at 1.6, it actually is `def` (not `(Just def)`) for everything here, so this is wrong.

Reply:
Basically, if we're not using the generic instance it seems surprising